baybe/recommenders/pure/base.py

```diff
@@ -31,6 +31,20 @@ class PureRecommender(ABC, RecommenderProtocol):
     """Allow to make recommendations that were measured previously.
     This only has an influence in discrete search spaces."""
 
+    user_callable: Optional[
+        Callable[
+            [
+                "PureRecommender",
+                Optional[SearchSpace],
+                Optional[int],
+                Optional[pd.DataFrame],
+                Optional[pd.DataFrame],
+            ],
+            Any,
+        ]
+    ] = field(default=None, kw_only=True)
+    """A user defined callable to process SearchSpace, int, and pandas DatFrames"""
```

Comment:
This should be more specific, since the `int` and the `DataFrames` have a special meaning here: They are the batch size and the training data.

Reply:
Also, it needs to be clear where exactly this function is called.

Reply:
I reformated the description, thanks!